R/thumbnail.R

```diff
@@ -0,0 +1,233 @@
+      path <- fs::file_temp(pattern = "content_image_", ext = ext)
+    } else {
+      # try png
+      warning(glue::glue("Could not infer file extension from content type: {ct}. Using '.png'"))
+      path <- fs::file_temp(pattern = "content_image_", ext = ".png")
+    }
+  }
+
+  writeBin(httr::content(res, as = "raw"), path)
+
+  return(fs::as_fs_path(path))
+}
+
+#' @rdname get_thumbnail
+#' @export
+get_image <- function(content, path = NULL) {
+  lifecycle::deprecate_warn("0.3.1", "get_image()", "get_thumbnail()")
+  get_thumbnail(content, path)
+}
+
+#' @rdname get_thumbnail
+#' @export
+delete_thumbnail <- function(content, path = NULL) {
+  validate_R6_class(content, "Content")
+  guid <- content$get_content()$guid
+
+  con <- content$get_connect()
+
+  if (!is.null(path)) {
+    scoped_experimental_silence()
+    get_thumbnail(content, path)
+  }
+
+  # Connect 2024.09.0 introduced public endpoints for content thumbnails. We
+  # prefer those, falling back to the unversioned endpoints if unavailable.
+  res <- con$DELETE(
+    v1_url("content", guid, "thumbnail"),
+    parser = NULL
+  )
+  if (httr::status_code(res) == 404) {
+    res <- con$DELETE(
+      unversioned_url("applications", guid, "image"),
+      parser = NULL
+    )
+  }
+  con$raise_error(res)
+
+  return(content)
+}
+
+#' @rdname get_thumbnail
+#' @export
+delete_image <- function(content, path = NULL) {
+  lifecycle::deprecate_warn("0.3.1", "delete_image()", "delete_thumbnail()")
+  delete_thumbnail(content, path)
+}
+
+#' @rdname get_thumbnail
+#' @export
+has_thumbnail <- function(content) {
+  validate_R6_class(content, "Content")
+  guid <- content$get_content()$guid
+
+  con <- content$get_connect()
+
+  # Connect 2024.09.0 introduced public endpoints for content thumbnails. We
+  # prefer those, falling back to the unversioned endpoints if unavailable.
+  res <- con$GET(
+    v1_url("content", guid, "thumbnail"),
+    parser = NULL
+  )
+  if (httr::status_code(res) == 404) {
+    res <- con$GET(
+      unversioned_url("applications", guid, "image"),
+      parser = NULL
+    )
+  }
+  con$raise_error(res)
+  httr::status_code(res) != 204
+}
+
+#' @rdname get_thumbnail
+#' @export
+has_image <- function(content) {
+  lifecycle::deprecate_warn("0.3.1", "has_image()", "has_thumbnail()")
+  has_thumbnail(content)
+}
+
+#' Set the Content Thumbnail
+#'
+#' Set the Content thumbnail using a variety of methods.
+#'
+#' @param content A content object
+#' @param path A file path or URL to an image
+#'
+#' @family content functions
+#' @rdname set_thumbnail
+#' @export
+set_thumbnail <- function(content, path) {
+  validate_R6_class(content, "Content")
+
+  valid_path <- NULL
+  if (file.exists(path)) {
+    valid_path <- path
+  } else {
+    parsed <- httr::parse_url(path)
+    if (parsed$scheme %in% c("http", "https")) {
+      valid_path <- fs::file_temp(pattern = "image", ext = fs::path_ext(parsed[["path"]]))
+      httr::GET(url, httr::write_disk(valid_path))
```

Comment:
What happens if this returns 404?

Reply:
Bad stuff — it just sets the image to (I assume) the 404 response, interpreted as a binary image. Thanks.